File: client/src/helpers/parse.js
export const parseImport = (source, wdDelim, cardDelim) => {
	const cards = source.split(cardDelim);
	const results = [];
	for (const card of cards) {
		const word = card.split(wdDelim)[0];
		const defi = card.split(wdDelim)[1];
		results.push({ word, definition: defi });
	}
	return results;
};
export const parseExport = (source, wdDelim, cardDelim) => {
	let result = '';
	for (const card of source) {
		const col = card.word + wdDelim + card.definition;
		result += col + cardDelim;
	}
	// remove last card delim
	result = result.slice(0, -cardDelim.length);
	return result;
};

function main() {
	let sample_input = 'word1 def1\nword2 def2\nword3 def3';
	let result1 = parseImport(sample_input, ' ', '\n');
	console.log(result1);
	let result2 = parseExport(result1, ' ', '\n');
	console.log(result2);
}

File: helpers/parse.py
# import
def parse_import(source: str, wd_delim: str, card_delim: str):
    cards = source.split(card_delim)
    results = []
    for card in cards:
        word = card.split(wd_delim)[0]
        defi = card.split(wd_delim)[1]
        results.append({"word": word, "definition": defi})
    return results

# export


def parse_export(source: list[dict], wd_delim: str, card_delim: str):
    result = ""
    for card in source:
        col = card["word"]+wd_delim+card["definition"]
        result += col + card_delim
    # remove last card delim
    result = result.rstrip(card_delim)
    return result


def app():
    sample_input = "word1 def1\nword2 def2\nword3 def3"
    result1 = parse_import(sample_input, " ", "\n")
    print(result1)
    result2 = parse_export(result1, " ", "\n")
    print(result2)


if __name__ == "__main__":
    app()
